firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -50,33 +80,45 @@ void moveDiagonal(int dirX, int dirY, int spacesX, int spacesY)
   digitalWrite(xMotor[1], dirX);
   digitalWrite(yMotor[1], dirY);
   
-  if (numStepsX = numStepsY) 
+  if (numStepsX == numStepsY) 
   {
     for(i = 0; i < numStepsX; i++)
     {
-    digitalWrite(xMotor[0], LOW);
-    digitalWrite(yMotor[0], LOW);
-    delay(1);
-    digitalWrite(xMotor[0], HIGH);
-    digitalWrite(yMotor[0], HIGH);
+      digitalWrite(xMotor[0], LOW);
+      digitalWrite(yMotor[0], LOW);
+      delay(1);
+      digitalWrite(xMotor[0], HIGH);
+      digitalWrite(yMotor[0], HIGH);
     }
   } 
 
-  else 
+  else if (numStepsY > numStepsX && (numStepsY / numStepsX) == 2)
   {
-    float dist = sqrt(numStepsX * numStepsX + numStepsY * numStepsY);
-    float delayX = dist / numStepsX;
-    float delayY = dist / numStepsY;
+    //can I call setScale here even thought it's declared after?
+    //This should be a slope of 2
+    setScale(xMotor, 2);
 
-    for(i=0; i < dist; i++)
+    for(i = 0; i < numStepsX; i++)
+    {
       digitalWrite(xMotor[0], LOW);
-      //do i have to convert this into an int, can i keep it as a variable?
-      delay(delayX);
+      digitalWrite(yMotor[0], LOW);
+      delay(1);
       digitalWrite(xMotor[0], HIGH);
+      digitalWrite(yMotor[0], HIGH);
+    }
+  }
+  else if (numStepsY < numStepsX)
```

Comment:
Just to be safe, I would add the same check for half like you did earlier.
Keep in mind, you have integer math here, so we can account for that by reversing x and y in the calculation.

&& (numStepsX / numStepsY) == 2

Reply:
If we don't have this we might end up in a situation where we're moving at a different slope.

Reply:
However, we may end up just using numStepsY > numStepsX and an else provided we only call this function with diagonal moves that have those slopes. We can adjust that later with Nashir.